app/src/main/java/de/westnordost/streetcomplete/data/upload/Uploader.kt

```diff
@@ -27,7 +27,7 @@ class Uploader(
 
     private val listeners = Listeners<UploadProgressSource.Listener>()
 
-    private val bannedInfo by lazy { versionIsBannedChecker.get() }
+    private lateinit var bannedInfo: BannedInfo
```

Comment:
`lateinit` always looks like a hack in Kotlin to me. Is it not possible to keep it `Lazy` or use a similar data structure? Is there a suspending Lazy?

Reply:
Agreed, I tried to find something nicer but this was the best I could find. I'm not a Kotlin person, so perhaps my research wasn't good enough 😞.

Happy to replace with anything you find!

Reply:
Hmm... I think [`Deferred`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-deferred/) with a `[CoroutineStart.LAZY](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-coroutine-start/-l-a-z-y/index.html)` might be it, didn't try it out though.

Or maybe we are overthinking it, and it could be just a nullable field.

Reply:
On second though, I am not sure about it... using GlobalScope is certainly not correct there... but without it, it would need to be suspending but `suspend val bannedInfo` is not possible as far as I know.

But no matter, if you are not a Kotlin person, I will look into it myself. As far as your work is concerned, the PR is complete and approved.

Reply:
I've looked a bit more. Including the Kotlin documentation doesn't mention anything about lateinit var should be preferred only for legacy APIs or something like that, so I guess it is fine after all, especially there does not exist a proper suspending lazy.